From 23dda8d39c718a9c58c860f989180a13778d9bcc Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.krummenacher@toradex.com>
Date: Wed, 11 Aug 2021 19:42:46 +0200
Subject: [PATCH 2/3] ARM: dts: imx6ull-colibri: use static MTD partition
 layout

For the recovery U-Boot we neither have eMMC nor NAND support in
U-Boot. Thus the fdt fixup for the MTD partitions does not work.

Use a static MTD partition layout so that we are not dependent
on U-Boot providing proper MTD partition information.

Signed-off-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 .../arm/boot/dts/nxp/imx/imx6ull-colibri.dtsi | 23 +++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/arch/arm/boot/dts/nxp/imx/imx6ull-colibri.dtsi b/arch/arm/boot/dts/nxp/imx/imx6ull-colibri.dtsi
index ec3c1e7301f4..1a29b5137069 100644
--- a/arch/arm/boot/dts/nxp/imx/imx6ull-colibri.dtsi
+++ b/arch/arm/boot/dts/nxp/imx/imx6ull-colibri.dtsi
@@ -157,6 +157,8 @@ ethphy1: ethernet-phy@2 {
 
 /* NAND */
 &gpmi {
+	#address-cells = <1>;
+	#size-cells = <1>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_gpmi_nand>;
 	fsl,use-minimum-ecc;
@@ -165,6 +167,27 @@ &gpmi {
 	nand-ecc-strength = <8>;
 	nand-ecc-step-size = <512>;
 	status = "okay";
+
+	partition@0 {
+		label = "mx6ull-bcb";
+		reg = <0x00000000 0x00080000>;
+	};
+	partition@80000 {
+		label = "u-boot1";
+		reg = <0x00080000 0x00180000>;
+	};
+	partition@200000 {
+		label = "u-boot2";
+		reg = <0x00200000 0x00180000>;
+	};
+	partition@380000 {
+		label = "u-boot-env";
+		reg = <0x00380000 0x00080000>;
+	};
+	partition@400000 {
+		label = "ubi";
+		reg = <0x00400000 0x1fc00000>;
+	};
 };
 
 /* I2C3_SDA/SCL on SODIMM 194/196 (e.g. RTC on carrier board) */
-- 
2.42.0

